### PR TITLE
Writing cohort banner

### PR DESCRIPTION
**Pull request by corwintines**
## Description
- Adds a DismissableBanner to promote the ethereum.org writing cohort

### CODE REVIEW

File: src/components/Banners/Implementations/WritersCohortBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React from "react"
+import { Text } from "@chakra-ui/react"
+
+import DismissableBanner from "../DismissableBanner"
+import Link from "../../Link"
+
+const WritersCohortBanner = () => {
+  if (
+    typeof window !== `undefined` &&
+    window.location.pathname.includes("contributing")
```
Comment (pettinarip) [inline src/components/Banners/Implementations/WritersCohortBanner.tsx:10]: For SSR purposes, a better way of getting the `pathname` is from Gatsby itself (or specifically from reach router which is what they use underneath).

```tsx
// every page or template receives a `location` prop
const StaticPage = ({ data, location }) => {
  ...
}

// then you can pass that prop down to the components
<WritersCohortBanner pathname={location.pathname} />
```

This way you abstract yourself from the `window` object and you can render the same thing server and client side.

**Review by pettinarip — APPROVED** [review]
Looks good! I left a comment for improvement.

File: src/components/Banners/Implementations/WritersCohortBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  ) {
+    return (
+      <DismissableBanner storageKey="writersCohort">
+        <Text m={0}>
+          Join the first-ever ethereum.org Writers Cohort, in collaboration with
+          Taptive, starting May 11th.{" "}
```
Comment (minimalsm) [inline src/components/Banners/Implementations/WritersCohortBanner.tsx:16]: Can probably remove Taptive from here for simplicity. 

```suggestion
          Join the first-ever ethereum.org Writers Cohort, starting May 11th.{" "}
```

File: src/components/Banners/Implementations/WritersCohortBanner.tsx (hunk 1 above)
Comment (corwintines) [reply src/components/Banners/Implementations/WritersCohortBanner.tsx]: Makes sense, changed this PR over to this just now.

File: src/components/Banners/Implementations/WritersCohortBanner.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+interface IProps {
+  pathname: string
+}
+
+const WritersCohortBanner = ({ pathname }) => {
+  if (
+    typeof window !== `undefined` &&
+    (pathname.includes("contributing") || pathname.includes("community"))
+  ) {
```
Comment (pettinarip) [inline src/components/Banners/Implementations/WritersCohortBanner.tsx:15]: My idea was to completely remove the `window` dep in order to render it in both places. Or does it break?

```suggestion
  if (pathname.includes("contributing") || pathname.includes("community")) {
```

**Review by pettinarip — APPROVED** [review]

**Pull request merged**